rtlib/src/SepDisegni.cpp

```diff
@@ -0,0 +1,275 @@
+//
+// Created by Sammy Stampley on 11/12/19.
+//
+//
+// Created by Sammy Stampley on 10/3/19.
+//
+
+#include "SepDisegni.h"
+
+//Constructor for the SepDisegni Class
+SepDisegni::SepDisegni(){
+    Mat image_;
+    Mat watershededImage_;
+    vector<Mat> fragmentedImage_;
+}
+
+
+
+//This takes an Input Image and sets it into your private image member
+const void SepDisegni::SetImage(Mat input) {
+    image_ = input;
+}
+
+//This retrieves the array of fragmented images that are stored
+const vector<Mat> SepDisegni::GetFragmentedImage() {
+    return fragmentedImage_;
+}
+
+vector<Mat> SepDisegni::Compute(){
+    watershededImage_ = watershed_image();
+    fragmentedImage_ = split_merge(watershededImage_);
+    return fragmentedImage_;
+}
+
+Mat SepDisegni::watershed_image() {
+
+
+    Mat src = image_;
+
+    // Show source image
+    namedWindow("Source Image", 0);
+    imshow("Source Image", src);
+    waitKey(0);
+
+
+    // Change the background from white to black, since that will help later to extract
+    // better results during the use of Distance Transform
+    for (int i = 0; i < src.rows; i++) {
+        for (int j = 0; j < src.cols; j++) {
+            if (src.at<Vec3b>(i, j) == Vec3b(255, 255, 255)) {
+                src.at<Vec3b>(i, j)[0] = 0;
+                src.at<Vec3b>(i, j)[1] = 0;
+                src.at<Vec3b>(i, j)[2] = 0;
+            }
+        }
+    }
```

Comment:
This is a curious thing to do and I feel like might be messing with our results.

Use descriptive loop variables. In this case, `i` should be `y` and `j` should be `x`.

Reply:
The source code for implementing the watershed method recommended using Distance Transform before sending the Mat image through the watershed function